src/ImageSharp/Formats/Webp/Lossless/Vp8LEncoder.cs

```diff
@@ -19,6 +19,15 @@ namespace SixLabors.ImageSharp.Formats.Webp.Lossless
     /// </summary>
     internal class Vp8LEncoder : IDisposable
     {
+        /// <summary>
+        /// Scratch buffer to reduce allocations.
+        /// </summary>
+        private readonly int[] scratch = new int[256];
```

Comment:
Instead of the `int[]` one could use fixed buffer wrapped in a private struct. See [this example](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA+ABATARgLABQGADAAQY4B0ASgK4B2ALgJYC2MlAkkzFBAA4BlXgDdmYGAGcA3IUIYAzOSykAwqQDehUjtLMeUegEMANqQaSjAMxjLSANX4AOADIBRepAAmvbbq0EukGk/FDMIkaMtoJgUJFgABakkrHxSQC8pPQwAO4AFACUsoHBOn6liuQopMKMefqMevQ+CAA0TY0RJrQwBaTpAHzJqYyJANr6LQC6/aRdPcWlOpUY1aomMEZQ9UzJjFuM7Q2kG/QA5owJfYPDcaMJlACCkoL8RvR5kvtQhycw55cCpR1pttkVysEIUFQuFIrYvlBaGBGjE7okof4MaUYREoqQrMwEDAvB1SMBaFYbFAxlgAKxIKaLJZBLHBSqwKyky7MSQTXaTGAIGY3DlkilUvnTJnMsolGXkJSvd4AHgaQ2eSo+xy+ByOu1OFyurKCAXlpVF2u+jGoME5mVF3N5Op+jONSwwAHZSABZGCsaAAT29W0kCVMwNgcM1eVFzuttvaBsB0plAF9Wem5aRM6mgA=).
(Note: the encapsulating class must be `unsafe` in order to make it work)

So `scratch` becomes a strong type and can have methods too 😄 
It avoids the extra array allocation, but at the cost of a bigger `Vp8LEncoder` -- I don't know what is better.

For code elegance I'd prefer the fixed buffer / strong type approach, as there could be a method like `Span<int> AsSpan(int start, int length, bool clear = true)` and that like.

Reply:
Or rent this array from an array pool / memory pool?
(Maybe that overkill here?)

Reply:
I think the array is too small to utilize the `MemoryAllocator` here.

On the fixed buffer wrapped in a private struct idea, I am undecided. @JimBobSquarePants what do you think should we go that way or should i leave it as it is?

Reply:
We use that trick in the jpeg decoder so there's precedence. The input to the enclosing type is always sanitized since your encoding so I'd go for it if you have the time. 

https://github.com/SixLabors/ImageSharp/blob/d0212221622b015706ca4d0728c269ec3caa5c7a/src/ImageSharp/Formats/Jpeg/Components/Decoder/HuffmanTable.cs#L13-L56

Reply:
mhm `MemoryMarshal.CreateSpan` does not exist in `net472`, `netstandard1.3` and `netstandard2.0`. What would be the right way to get the Span there? I think, i miss something obvious.

Reply:
Ah, that's too bad. One would need to create the span via the pointer overload, but that needs a fixed-block which isn't feasible here (? or is it).

But: do we need to create a span at all? The span is needed to call Clear, and use indexed access to the elements. This can be done in the `Scratch`-type / buffer directly. So the buffer could be passed around instead of the span.
Note: the buffer is now a type, so can have methods too.

In order to clear effeciently for .NET Core+ one could create a span and call clear (as it's vectorized), for older runtimes just use a manual loop (maybe unrolled).
Or even better just rely on the runtime to zero-init (iif the `[SkipLocalsInit]` isn't applied, which isn't at the moment) the fixed sized buffer, so instead of clearing create a new buffer.

I'll think about this a bit more, if there is maybe any better approach to that.

Reply:
I tried to change it to a struct, but without being able to create a Span from `MemoryMarshal.CreateSpan` it seems to make things more complicated then before. For example all methods in `PredictorEncoder` operate on Spans.
Maybe there is a way of doing it without overcomplicating things, but i dont see it.

I would like to keep it as it is for now.